### PR TITLE
rna(chore): update DefaultContent, migrate SignUp

### PR DESCRIPTION
**Pull request by calebpollman**
#### Description of changes
Update `DefaultContent`, migrate `SignUp` to use `DefaultContent`

#### Checklist

- [ ] PR description included
- [ ] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx [hunk 1]
```diff
@@ -45,26 +34,34 @@ const SignUp: DefaultSignUpComponent = ({
...
+      primary: { children: primaryButtonText, onPress: handleFormSubmit },
+      links: hideSignIn
+        ? undefined
+        : [{ children: secondaryButtontext, onPress: toSignIn }],
+    }),
+    [
+      handleFormSubmit,
+      hideSignIn,
+      primaryButtonText,
+      secondaryButtontext,
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx:53]:
```suggestion
      secondaryButtonText,
```

File: packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx [hunk 2]
```diff
@@ -45,26 +34,34 @@ const SignUp: DefaultSignUpComponent = ({
...
+    ? getCreatingAccountText()
+    : getCreateAccountText();
+  const secondaryButtontext = getSignInTabText();
+
+  const buttons = useMemo(
+    () => ({
+      primary: { children: primaryButtonText, onPress: handleFormSubmit },
+      links: hideSignIn
+        ? undefined
+        : [{ children: secondaryButtontext, onPress: toSignIn }],
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx:47]:
```suggestion
        : [{ children: secondaryButtonText, onPress: toSignIn }],
```

File: packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx [hunk 3]
```diff
@@ -45,26 +34,34 @@ const SignUp: DefaultSignUpComponent = ({
     handleSubmit,
   });
 
+  const primaryButtonText = isPending
+    ? getCreatingAccountText()
+    : getCreateAccountText();
+  const secondaryButtontext = getSignInTabText();
```
Comment (ioanabrooks) [inline packages/react-native/src/Authenticator/Defaults/SignUp/SignUp.tsx:40]:
```suggestion
  const secondaryButtonText = getSignInTabText();
```

**Review by ioanabrooks — COMMENTED** [review]
Just a small nit

**Review by ioanabrooks — APPROVED** [review]

**Review by dbanksdesign — APPROVED** [review]

**Pull request merged**